### PR TITLE
Update world events

### PR DESCRIPTION
**Pull request by florensie**
This is still untested, but open for reviews.
- Only updated existing patchwork events, no new ones were added
- `WorldTypeEvent` was removed upstream
- `DifficultyChangeEvent` implementation was broken upstream, but now fixed and implemented here as well

### CODE REVIEW

File: patchwork-events-world/src/main/java/net/minecraftforge/event/world/WorldEvent.java [hunk 1]
```diff
@@ -159,35 +158,25 @@ public BlockPos getPos() {
...
 
-		public LevelInfo getSettings() {
+		public ServerWorldProperties getSettings() {
 			return settings;
 		}
-
-		@Override
-		public boolean isCancelable() {
-			return true;
-		}
```
Comment (WalkerKnapp) [inline patchwork-events-world/src/main/java/net/minecraftforge/event/world/WorldEvent.java]: This event is still cancellable in forge, this should be readded.

File: patchwork-events-world/src/main/java/net/minecraftforge/event/world/WorldEvent.java [hunk 2]
```diff
@@ -159,35 +158,25 @@ public BlockPos getPos() {
...
 
-		public List<SpawnEntry> getList() {
+		public List<SpawnSettings.SpawnEntry> getList() {
 			return list;
 		}
-
-		@Override
-		public boolean isCancelable() {
-			return true;
-		}
```
Comment (WalkerKnapp) [inline patchwork-events-world/src/main/java/net/minecraftforge/event/world/WorldEvent.java]: This event is still cancellable in forge, this should be readded.

**Review by WalkerKnapp — CHANGES_REQUESTED** [review]
I just wrote up a test mod, everything works well.  PotentialSpawns and CreateSpawnPosition just need to be marked cancellable.

File: patchwork-events-world/src/main/java/net/minecraftforge/event/world/WorldEvent.java (hunk 1 above)
Comment (florensie) [reply patchwork-events-world/src/main/java/net/minecraftforge/event/world/WorldEvent.java]: Don't know how I missed that.

**Review by TheGlitch76 — APPROVED** [review]

**Review by WalkerKnapp — APPROVED** [review]

**Pull request merged**